apps/condo/domains/banking/components/BankAccountReport.tsx

```diff
@@ -137,8 +137,10 @@ const BankAccountReportContent: IBankReportContent = ({ bankAccountReports = [],
     const { breakpoints, isMobile } = useLayoutContext()
     const router = useRouter()
 
+    const initialSelectValue = bankAccountReports.map(({ period }) => period).indexOf(get(router, 'query.period'))
```

Comment:
Here you rely on array indexes, that aren't reliable actually. Please either rename value to reflect that it's index, not value itself, or change logic to rely on real values.